src/Tasks/DownloadFile.cs

```diff
@@ -146,7 +146,8 @@ private async Task<bool> ExecuteAsync()
         private async Task DownloadAsync(Uri uri, CancellationToken cancellationToken)
         {
             // The main reason to use HttpClient vs WebClient is because we can pass a message handler for unit tests to mock
-            using (var client = new HttpClient(HttpMessageHandler ?? new HttpClientHandler(), disposeHandler: true) { Timeout = TimeSpan.FromMilliseconds(Timeout) })
+            using var httpHandler = new HttpClientHandler();
```

Comment:
Seems like there are now circumstances where this object would be created but not used, are those bad and should we avoid them?